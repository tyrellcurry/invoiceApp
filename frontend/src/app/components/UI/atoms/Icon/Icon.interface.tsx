export interface IIconProps {
  className?: string;
  name: 'dm-moon' | 'dm-sun';
  fill?: string;
}
