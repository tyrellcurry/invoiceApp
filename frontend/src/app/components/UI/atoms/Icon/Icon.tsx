/**
 * @name Icon
 * @author Tyrell Curry <tyrellcurryio@gmail.com>
 *
 * Used for rendering icons
 *
 * @param name - name of icon set in IIconProps type
 * @param props - className
 *
 * @returns {JSX.Element}
 */

import React, { JSX } from 'react';
import { IIconProps } from '@/app/components/UI/atoms/Icon/Icon.interface';
import Moon from '@/app/components/UI/atoms/Icon/assets/dm-moon.svg';
import Sun from '@/app/components/UI/atoms/Icon/assets/dm-sun.svg';

const Icon = ({ name, ...props }: IIconProps): JSX.Element => {
  let ComponentToRender: React.ElementType | null = null;

  switch (name) {
    case 'dm-moon':
      ComponentToRender = Moon;
      break;
    case 'dm-sun':
      ComponentToRender = Sun;
      break;
    default:
      break;
  }

  return <>{ComponentToRender && <ComponentToRender {...props} />}</>;
};

export default Icon;
