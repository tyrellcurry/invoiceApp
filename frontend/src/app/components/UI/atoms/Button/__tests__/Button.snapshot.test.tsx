import React from 'react';
import { render } from '@testing-library/react';
import Icon from '@/app/components/UI/atoms/Icon/Icon';

describe('Icon Component', () => {
  it('renders correctly with an icon name passed', () => {
    const { container } = render(<Icon name="dm-moon" />);
    expect(container).toMatchSnapshot();
  });
});
