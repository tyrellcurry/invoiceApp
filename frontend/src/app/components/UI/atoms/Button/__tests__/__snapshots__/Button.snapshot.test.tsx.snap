// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders correctly with an icon name passed 1`] = `
<div>
  <svg />
</div>
`;
